localgov_alert_banner.module

```diff
@@ -98,3 +101,61 @@ function localgov_alert_banner_theme_suggestions_localgov_alert_banner(array $va
   $suggestions[] = 'localgov_alert_banner__' . $entity->id() . '__' . $sanitized_view_mode;
   return $suggestions;
 }
+
+/**
+ * Implements hook_localgov_alert_banner_view().
+ */
+function localgov_alert_banner_localgov_alert_banner_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+
+  // Check if the field is present and has a value.
+  if (!$entity->hasField('field_conditions') || $entity->get('field_conditions')->isEmpty()) {
+    return TRUE;
+  }
+
+  // Context handler service.
+  $context_handler = \Drupal::service('context.handler');
+
+  $conditions = [];
+
+  // Get the conditions field value.
+  $conditions_config = $entity->get('field_conditions')->getValue()[0]['conditions'];
+
+  // Condition plugin manager.
+  $manager = \Drupal::service('plugin.manager.condition');
+
+  // Get the current route.
+  $route = \Drupal::routeMatch();
+  $route_name = $route->getRouteName();
+  $route_parameters = $route->getParameters();
+
+  // Bypass condition check if it is the alert banner route or confirmation.
+  $bypass_conditions = ($route_name == 'entity.localgov_alert_banner.canonical' || $route_name == 'entity.localgov_alert_banner.status_form');
+
+  if (!$bypass_conditions) {
+
+    // Loop through each condition.
+    foreach ($conditions_config as $condition_id => $values) {
+
+      // Add current node as context if required and on a node route.
+      $node = $route_parameters->get('node');
```

Comment:
This got me to thinking. The conditions make sense, hopefully they'll fit the requirements that the product group come up with selecting where things are shown. But for this part...

Would it make more sense for this to be in the block build() https://github.com/localgovdrupal/localgov_alert_banner/blob/1a30057581758a56ced975c8cb11c7d9b952589a/src/Plugin/Block/AlertBannerBlock.php#L122
Good because it's earlier and also here you could get things like the node by adding it to the block context
```
 *   context = {
 *     "node" = @ContextDefinition("entity:node")
 *   }
```
and pulling it `$node = $this->getContextValue('node');`
Which gives the added bonus that iirc the ContextAwarePlugin stuff does the cache contexts?

Reply:
That could work, would there be circumstances we render the entity outside the block? Would we assume users who are doing that actully know what they are doing so should handle it themselves.

Reply:
Also, this block may still exist on non node pages. Does adding a block context of node break that @ekes ?